RFC/src/RFC-0130_Mining.md

```diff
@@ -1 +1,92 @@
+
+### Full-node mining on Tari base layer
+
+The document is divided into two parts. First an overview will be provided describing the Tari merge mining process, then a descriptive list of the primary functionality required by the Tari full-node miner will be proposed.
+
+
+####  Overview of Tari merged mining process
+
+Valid transactions that need to be included on the Tari blockchain should be propagated by the mining full-nodes on the Tari base layer network. A Tari mining full-node should retrieve transactions from its mempool and assemble a new Tari block by bundling transactions together. It should ensure that the transactions that are included in the new Tari block are valid and that rules such as signatures, multi-signatures and timelocks are enforced before they are including in a new block.
+
+As Tari could be merged mined with Monero, both a Tari and a Monero block needs to be created and linked by including the Tari block information in the Monero block, and Monero block information in the Tari block. First, a new Tari block is created and then the block header hash of the new Tari block should be included in the coinbase transaction of the new Monero block. Once the block construction is complete, the mining full-node should perform the CryptoNight Proof-of-Work (PoW) algorithm on the Monero block that includes the information of the new Tari block. 
+
+The solution to the PoW problem could be solved at the difficulty of either the Tari and/or Monero blockchain. If a solution has been found that meets the minimum difficulty requirements of the Monero blockchain, then the new Tari block should be updated by including the solved Monero block’s information (block header hash, merkel tree branch, and hash of the coinbase transaction) into the PoW summary section of the Tari block header. If a solution was found that meet the Tari or Monero blockchain difficulty then the new Tari block can be added to the Tari blockchain. If the solution met the difficulty requirements of the Monero blockchain then the new Monero block can also be added to the Monero blockchain. If the PoW solution was sufficient to meet the difficult level of both the Tari and Monero blockchains then the individual blocks for each cryptocurrency can be added to their respective blockchains.
+
+Solved and completed blocks should be propagated to the rest of the base layer network so that Nakamoto consensus can be performed on the new block and it can be added to the local blockchain copies of the full mining nodes.
+
+####  Primary functionality required by a Tari full-node miner
+- The Tari blockchain should have the ability to be merged mined with Monero using the CryptoNight Proof-of-Work algorithm. 
```

Comment:
Could you work through this section again using RFC2119 conventions? This entails capitalising MUST, SHOULD etc., where those requirements are stated.

Given that this RFC assumes merge mining, this would surely be a MUST, right? i.e. there's no valid Tari node software that only mines Tari, correct?

Reply:
Suggested changes were implemented.